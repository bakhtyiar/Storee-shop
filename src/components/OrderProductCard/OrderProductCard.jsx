import React, {useContext, useEffect, useState} from "react";
import {RootContext} from "../../contexts/root-context/root-context";
import {Link} from "react-router-dom";
import {routes} from "../../utils/constants";
import imageIcon from "../../assets/img/icons/image/icons8-image-48.png";

const OrderProductCard = ({product, selfIndexInCart}) => {
    const {cartState: {onRemoveFromCart, onUpdateQuantity}} = useContext(RootContext);
    const [quantity, setQuantity] = useState(product.quantity);

    useEffect(() => {
        if (quantity < 1) {
            onRemoveFromCart(selfIndexInCart);
        } else {
            onUpdateQuantity(product.id, quantity);
        }
    }, [quantity]);

    return (
        <Link to={`${routes.product.path}/${product.id}`}
              className='px-2 py-3 d-flex flex-row align-items-center text-decoration-none text-black'>
            <div className='w-100 d-flex flex-row'>
                <div className={'bg-image rounded align-self-center'}
                     style={{
                         height: '128px',
                         width: '128px',
                         backgroundImage: `url(${product.thumbnail || imageIcon})`,
                         backgroundPosition: 'center',
                         backgroundSize: 'contain',
                         backgroundRepeat: 'no-repeat'
                     }}/>
                <div className='flex-column align-self-center ms-2'>
                    <h6 className='my-0'>{product.title}</h6>
                    {product.quantity && <p className='my-0'>Quantity: {product.quantity}</p>}
                    {product.category && <p className='my-0'>Brand: {product.category}</p>}
                    {product.brand && <p className='my-0'>Brand: {product.brand}</p>}
                    {product.rating && <p className='my-0'>Rating: {product.rating}</p>}
                </div>
            </div>
            <div className='ml-3 d-flex gap-2'>
                <h6>Price: ${product.price}</h6>
            </div>
        </Link>
    );
};

export default OrderProductCard;