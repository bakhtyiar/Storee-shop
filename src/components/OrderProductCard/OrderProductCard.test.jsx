import {screen} from "@testing-library/react";
import OrderProductCard from "./OrderProductCard";
import {renderWithRouter} from "../../tests/helpers/renderWithRouter";

describe("Basic tests", () => {
    test("Match snapshot", () => {
        renderWithRouter(<OrderProductCard product={{quantity: 1}} selfIndexInCart={0}/>);
        expect(screen.getByTestId("order-product-card")).toMatchSnapshot();
    })
})