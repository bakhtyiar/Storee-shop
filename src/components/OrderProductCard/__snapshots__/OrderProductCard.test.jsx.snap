// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic tests Match snapshot 1`] = `
<a
  class="px-2 py-3 d-flex flex-row align-items-center text-decoration-none text-black"
  data-testid="order-product-card"
  href="/product/undefined"
>
  <div
    class="w-100 d-flex flex-row"
  >
    <div
      class="bg-image rounded align-self-center"
      style="height: 128px; width: 128px; background-image: url(icons8-image-48.png); background-position: center; background-size: contain; background-repeat: no-repeat;"
    />
    <div
      class="flex-column align-self-center ms-2"
    >
      <h6
        class="my-0"
      />
      <p
        class="my-0"
      >
        Quantity: 
        1
      </p>
    </div>
  </div>
  <div
    class="ml-3 d-flex gap-2"
  >
    <h6>
      Price: $
    </h6>
  </div>
</a>
`;
